atat_internal_api.yaml

```
@@ -0,0 +1,2197 @@
+      properties:
+        clinNumber:
+          pattern: "(?!^0{4}$)^\\d{4}$"
+          type: string
+          description: "Contract Line Item Number (CLIN), 0001 through 9999"
+        obligatedFunds:
+          type: number
+        idiqClin:
+          type: string
+        popStartDate:
+          type: string
+          description: "Start of period of performance"
+          format: date
+        popEndDate:
+          type: string
+          description: "End of period of performance"
+          format: date
+        totalClinValue:
+          type: number
+      additionalProperties: false
+      description: "Represents a Clin in a Task Order"
+    TaskOrder:
+      required:
+        - clins
+        - taskOrderFile
+        - taskOrderNumber
+      type: object
+      allOf:
+        - $ref: "#/components/schemas/BaseObject"
+      properties:
+        clins:
+          minItems: 1
+          type: array
+          items:
+            $ref: "#/components/schemas/Clin"
+        taskOrderFile:
+          description: "Metadata associated with file which was previously uploaded\
+            \ by POST /taskOrderFiles"
+          allOf:
+            - $ref: "#/components/schemas/FileMetadataSummary"
+        taskOrderNumber:
+          pattern: "^[\\d]{13,17}$"
+          type: string
+          description: "TO numbers are 13 characters. TO modifications are 17 characters."
+      additionalProperties: false
+      description: "Represents a Task Order in the Funding Step"
+    FileMetadataSummary:
+      required:
+        - id
+        - name
+      type: object
+      properties:
+        name:
+          type: string
+          description: "name of file when uploaded"
+        id:
+          type: string
+          description: "Should be a UUID. Will be the S3 Object Key."
+      additionalProperties: false
+      description: "Metadata describing an uploaded file"
+    PortfolioAccess:
+      description: "Represents a set of Operators who should be granted access to\
+        \ a Portfolio at a specific access level. Uses an enum construct for forward\
+        \ compatibility should other access levels be specified."
+      properties:
+        portfolioAdministrators:
+          type: array
+          items:
+            type: string
+            format: email
+    AppEnvAccess:
+      description: "Represents a set of Operators who should be granted access to\
+        \ an Application or Environment at a specific access level. The same access\
+        \ levels are available for both Applications & Environments."
+      properties:
+        administrators:
+          type: array
+          items:
+            type: string
+            format: email
+        contributors:
+          type: array
+          items:
+            type: string
+            format: email
+        readOnlyOperators:
+          type: array
+          items:
+            type: string
+            format: email
```

```suggestion
    AppEnvAccess:
      description: "Represents a set of Operators who should be granted access to\
        \ an Application or Environment at a specific access level. The same access\
        \ levels are available for both Applications & Environments."
      required:
          - administrators
          - contributors
          - readOnlyOperators
      properties:
        administrators:
          type: array
          items:
            type: string
            format: email
        contributors:
          type: array
          items:
            type: string
            format: email
        readOnlyOperators:
          type: array
          items:
            type: string
            format: email
```
Are any of these properties required? If so:
I suggest adding a required field validation check, so the frontend knows to match the `AppEnvAccess` properties of adminitrators, contributors, and readOnlyOperators. They could send empty arrays, which is probably fine, but if it isn't fine I recommend adding a `minItems: 1` to the arrays.

---

I don't think we need to make them required on the API side; we just assume that there are no operators with that particular access level if we never get them in a PATCH.

---

These aren't required so let's leave it as-is for now. Plus we may end up redesigning this depending on how the discussions go next week.